assets/wizards/popups/views/campaign-group-management/index.js

```diff
@@ -0,0 +1,104 @@
+/**
+ * Campaign Groups management.
+ */
+
+/**
+ * WordPress dependencies.
+ */
+import apiFetch from '@wordpress/api-fetch';
+import { useEffect, useState, Fragment } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Material UI dependencies.
+ */
+import DeleteIcon from '@material-ui/icons/Delete';
+
+/**
+ * Internal dependencies
+ */
+import { withWizardScreen, ActionCard, Button, TextControl } from '../../../../components/src';
+import './style.scss';
+
+/**
+ * Popup group screen.
+ */
+const CampaignGroupManagement = ( {} ) => {
+	const [ groups, setGroups ] = useState( -1 );
+	const [ inFlight, setInFlight ] = useState( false );
+	const [ newGroupName, setNewGroupName ] = useState( '' );
+
+	const createTerm = term => {
+		setInFlight( true );
+		apiFetch( {
+			path: '/wp/v2/newspack_popups_taxonomy',
+			method: 'POST',
+			data: {
+				name: term,
+				slug: term,
+			},
+		} ).then( () => {
+			setInFlight( false );
+			retrieveTerms();
+		} );
+	};
+
+	const deleteTerm = id => {
+		apiFetch( {
```

Comment:
Suggestion: set `inFlight` to block UI when the request is made.

Reply:
Done in 8baca9e